src/bms/player/beatoraja/play/GrooveGauge.java

```diff
@@ -308,13 +309,26 @@ public float modify(float f, BMSModel model) {
 					}else{
 						fix1=10.0f;
 					}
-					int note=1000;
-					float mod=0.002f;
-					while(note>model.getTotalNotes()||note>1){
-						fix2 += mod * (float)(note - Math.max(model.getTotalNotes(), note/2));
-						note/=2;
-						mod*=2.0f;
+
+					// ノート数補正 (<1000)
+					if(model.getTotalNotes()<=20) {
+						fix2 = 10.0f;
+					}else if(model.getTotalNotes()<30) {
+						fix2 = 8.0f + 0.2f*(30-model.getTotalNotes());
+					}else if(model.getTotalNotes()<60) {
+						fix2 = 5.0f + 0.2f*(60-model.getTotalNotes())/3.0f;
+					}else if(model.getTotalNotes()<125) {
+						fix2 = 4.0f + (125-model.getTotalNotes())/65.0f;
+					}else if(model.getTotalNotes()<250) {
+						fix2 = 3.0f + 0.008f*(250-model.getTotalNotes());
+					}else if(model.getTotalNotes()<500) {
+						fix2 = 2.0f + 0.004f*(500-model.getTotalNotes());
+					}else if(model.getTotalNotes()<1000) {
```

Comment:
You should probably put output of `model.getTotalNotes()` into a temporary variable so that it doesn't get called 7 times in a row.